pages/platform/kubernetes-k8s/backup-and-restore-cluster-namespace-and-applications-with-trilio/guide.en-us.md

```diff
@@ -0,0 +1,1275 @@
+
+Steps to create a `Target` for `TrilioVault`:
+
+1. First, change directory where the `ovh/docs` Git repository was cloned on your local machine:
+
+    ```shell
+    cd docs/pages/platform/kubernetes-k8s/backup-and-restore-cluster-namespace-and-applications-with-trilio/
+    ```
+
+2. Next, create the Kubernetes secret containing your target S3 bucket credentials (please replace the `<>` placeholders accordingly):
+
+    ```shell
+    kubectl create secret generic trilio-ovh-s3-target \
+      --namespace=tvk \
+      --from-literal=accessKey="<YOUR_OVH_OBJECT_STORAGE_BUCKET_ACCESS_KEY_HERE>" \
+      --from-literal=secretKey="<YOUR_OVH_OBJECT_STORAGE_BUCKET_SECRET_KEY_HERE>"
+    ```
+
+3. Then, open and inspect the `Target` manifest file provided in the `docs` repository, using an editor of your choice (preferably with `YAML` lint support). You can use [VS Code](https://code.visualstudio.com) for example:
+
+    ```shell
+    cat assets/manifests/triliovault-ovh-s3-target.yaml
+    ```
+
+4. Now, please replace the `<>` placeholders accordingly for your OVH Object Storage `Trilio` bucket, like: `bucketName`, `region`,  `url` and `credentialSecret`.
+5. Finally, save the manifest file and create the `Target` object using `kubectl`:
+
+    ```shell
+    kubectl apply -f assets/manifests/triliovault-ovh-s3-target.yaml
+    ```
+
+What happens next is, `TrilioVault` will spawn a `worker job` named `trilio-ovh-s3-target-validator` responsible with validating your S3 bucket (like availability, permissions, etc.). If the job finishes successfully, the bucket is considered to be healthy or available and the `trilio-ovh-s3-target-validator` job resource is deleted afterwards. If something bad happens, the S3 target validator job is left up and running so that you can inspect the logs and find the possible issue.
+
+Now, please go ahead and check if the `Target` resource created earlier is `healthy`:
+
+```shell
+kubectl get target trilio-ovh-s3-target -n tvk
+```
+
+The output looks similar to (notice the `STATUS` column value - should be `Available`, meaning it's in a `healthy` state):
+
+```text
+NAME                   TYPE          THRESHOLD CAPACITY   VENDOR   STATUS      BROWSING ENABLED
+trilio-ovh-s3-target   ObjectStore   10Gi                 Other    Available   Enabled
+```
+
+If the output looks like above, then you configured the S3 target object successfully.
+
+**Hint:**
+
+In case the target object fails to become healthy, you can inspect the logs from the `trilio-ovh-s3-target-validator` Pod to find the issue:
+
+First, you need to find the target validator
+
+	```shell
+	kubectl get pods -n tvk | grep trilio-ovh-s3-target-validator
+	```
+
+The output looks similar to:
+
+	```text
+	trilio-ovh-s3-target-validator-tio99a-6lz4q	1/1     Running     0          104s
+	```
```

Comment:
I'm blocked here, don't know why
`trilio-ovh-s3-target-validator-66a65q--1-dflmn                 0/1     Error              0              5m31s`
Logs are similar to those presented below:

```
...
INFO:root:waiting for command to complete...
2022-01-05 16:30:24.013 12 WARNING s3 [-] Invalid S3 SSL cert file path: /etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pemUsing Default path, set the value to False for insecure connection
Exception in thread Thread-1:
Traceback (most recent call last):
  File "/usr/lib64/python3.6/threading.py", line 919, in _bootstrap_inner
    self.run()
  File "/usr/lib64/python3.6/threading.py", line 867, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/local/lib/python3.6/site-packages/s3fuse/s3vaultfuse.py", line 2325, in run
    s3backend.validate_s3_client(options)
  File "/usr/local/lib/python3.6/site-packages/s3fuse/vaults3.py", line 103, in validate_s3_client
    self.__client.head_bucket(Bucket=self.__bucket_name)
  File "/usr/local/lib/python3.6/site-packages/botocore/client.py", line 391, in _api_call
    return self._make_api_call(operation_name, kwargs)
  File "/usr/local/lib/python3.6/site-packages/botocore/client.py", line 719, in _make_api_call
    raise error_class(parsed_response, operation_name)
botocore.exceptions.ClientError: An error occurred (400) when calling the HeadBucket operation: Bad Request

Mounting...
Mounted....
....
....
INFO:root:2022-01-05 16:32:22.587795: waiting for mount operation to complete.
ERROR:root:2022-01-05 16:32:24.590235: timeout exceeded, not able to mount within time.
ERROR:root:/triliodata is not a mountpoint. We can't proceed further.
Traceback (most recent call last):
  File "/opt/tvk/datastore-attacher/mount_utility/mount_by_target_crd/mount_datastores.py", line 56, in main
    utilities.mount_datastore(metadata, datastore.get(constants.DATASTORE_TYPE), base_path)
  File "/opt/tvk/datastore-attacher/mount_utility/utilities.py", line 377, in mount_datastore
    mount_s3_datastore(metadata_list, base_path)
  File "/opt/tvk/datastore-attacher/mount_utility/utilities.py", line 306, in mount_s3_datastore
    wait_until_mount(base_path)
  File "/opt/tvk/datastore-attacher/mount_utility/utilities.py", line 328, in wait_until_mount
    base_path))
Exception: /triliodata is not a mountpoint. We can't proceed further.
```
Could it be an SSL issue ? An auth issue ? 
Everything is Ok with OVH S3 setup

Reply:
@raclepoulpe TVK is compatible with the Storage which is S3 compatible.
I have also faced this issue while I used the `OVH Object Store Container` for Target Repo for Backup storage. So, I have used the AWS S3 bucket with appropriate permission and it works for us.

I have raised this issue with Eileen Montiel in our email conversation. I need some help to understand how OVH's Object Store can be accessed using S3 Swift API. Once this is resolved, I will update the page and target yaml with those specific details.
For now, can you try to use either AWS S3 or any other S3 compatible storage or NFS Server as Target and proceed?

Reply:
@bhagirathhapse thank you for the information!
I'll try to use another storage so.

Reply:
@raclepoulpe After our meeting with Eileen and Charlette, I was able to create and use the `OVH Object Storage Container` as the TVK Target for backup storage. I have updated the instruction which also includes some articles from OVH documentation. Specifically [Getting Started with Swift S3 API](https://docs.ovh.com/gb/en/storage/getting_started_with_the_swift_S3_API/)
Do try and let me know if you face any issue which creating Target using OVH Object Storage Container.

Reply:
i still have the issue.
```
Mounting...
Mounted....
Exception in thread Thread-1:
Traceback (most recent call last):
  File "/usr/lib64/python3.6/threading.py", line 919, in _bootstrap_inner
    self.run()
  File "/usr/lib64/python3.6/threading.py", line 867, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/local/lib/python3.6/site-packages/s3fuse/s3vaultfuse.py", line 2325, in run
    s3backend.validate_s3_client(options)
  File "/usr/local/lib/python3.6/site-packages/s3fuse/vaults3.py", line 103, in validate_s3_client
    self.__client.head_bucket(Bucket=self.__bucket_name)
  File "/usr/local/lib/python3.6/site-packages/botocore/client.py", line 391, in _api_call
    return self._make_api_call(operation_name, kwargs)
  File "/usr/local/lib/python3.6/site-packages/botocore/client.py", line 719, in _make_api_call
    raise error_class(parsed_response, operation_name)
botocore.exceptions.ClientError: An error occurred (400) when calling the HeadBucket operation: Bad Request
...
...
ERROR:root:/triliodata is not a mountpoint. We can't proceed further.
Traceback (most recent call last):
  File "/opt/tvk/datastore-attacher/mount_utility/utilities.py", line 379, in mount_datastore
    mount_s3_datastore(metadata_list, base_path)
  File "/opt/tvk/datastore-attacher/mount_utility/utilities.py", line 307, in mount_s3_datastore
    wait_until_mount(base_path)
  File "/opt/tvk/datastore-attacher/mount_utility/utilities.py", line 329, in wait_until_mount
    base_path))
Exception: /triliodata is not a mountpoint. We can't proceed further.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/tvk/datastore-attacher/mount_utility/mount_by_target_crd/mount_datastores.py", line 56, in main
    utilities.mount_datastore(metadata, datastore.get(constants.DATASTORE_TYPE), base_path)
  File "/opt/tvk/datastore-attacher/mount_utility/utilities.py", line 382, in mount_datastore
    mount_s3_datastore(metadata_list, base_path)
  File "/opt/tvk/datastore-attacher/mount_utility/utilities.py", line 307, in mount_s3_datastore
    wait_until_mount(base_path)
  File "/opt/tvk/datastore-attacher/mount_utility/utilities.py", line 329, in wait_until_mount
    base_path))
Exception: /triliodata is not a mountpoint. We can't proceed further.
```

Reply:
@raclepoulpe I understand this problem since I have gone through this. I will fix all other issues on the PR.
I want to request if we can have a 30 min meeting to understand the TVK installation procedure. This will help you unblock and then you can validate backup/restore operations through yamls manifests or through TVK Management Console.
Let me know, if this plan is good for you and I will send meeting invite that can suit your timezone.